Sources/SwiftProtobuf/_MessageStorage.swift

```diff
@@ -220,12 +229,7 @@ extension _MessageStorage {
             }
         }
 
-        // Copy all of the trivial values (including has-bits) in bitwise fashion. Note that if we
-        // never set `firstNontrivialStorageOffset`, then the whole message must have been trivial
-        // fields. (This also works if the message is empty; that is, zero bytes.)
-        if firstNontrivialStorageOffset == 0 {
-            firstNontrivialStorageOffset = layout.size
-        }
+        // Copy all of the trivial values (including has-bits) in bitwise fashion.
         if firstNontrivialStorageOffset != 0 {
```

Comment:
Thinking some more about this, is it even possible for this to be zero now?  Seems like that could only happen with an empty message.

- If there's only complex fields (messages, strings, repeated), won't there still be has bits?
- If all the fields are in a oneof, no matter what the type of fields, won't there still be the storage for the "active" oneof item?

So maybe skip the gate since we don't need to optimize that case (before I was thinking all complex would skip this, but I forgot about the hasbits).